core/transientstore/delete_store_test.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package transientstore
+
```

Comment:
- Can you add one more test that mimics the crash-recovery (i.e., automatically delete at start)
- Can you see if you can cover some of the error paths easily by directly inserting some data into the db, or by closing the db provider at some occasions.
- I feel that some of the tests can be merged into one and have them as a table-driven sub-tests pattern.

Reply:
OK.

- `TestProviderRestartAfterFailedDeletionScrubsPendingDeletions` runs a simulated test of the crash recovery at peer (NewProvider) init. 
- The tests do insert some records into the transient storage - are there additional insertions / patterns in addition to the `populateTestStore()` ?